sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -175,6 +175,30 @@
  * <br>
  * 3. Start from <em>latest</em> offset by default;
  *
+ * <h3>Use Avro schema with Confluent Schema Registry</h3>
+ *
+ * <p>If the message keys or/and values in a topic were serialised by Avro schema stored in
+ * Confluent Schema Registry, KafkaIO source can fetch this schema from specified Schema Registry
+ * URL and use it for deserialization further. As an output, it will return a {@link PCollection} of
+ * {@link KafkaRecord}s where key or/and value will be typed as {@link
+ * org.apache.avro.generic.GenericRecord}. In this case, users don't need to specify key or/and
+ * value deserializers and coders since they will be set to {@link KafkaAvroDeserializer} and {@link
+ * AvroCoder} by default accordingly.
+ *
+ * <p>For example, below topic values are serialized with Avro schema stored in Schema Registry,
+ * keys are typed as {@link Long}:
+ *
+ * <pre>{@code
+ * PCollection<KafkaRecord<Long, GenericRecord>> input = pipeline
+ *   .apply(KafkaIO.<Long, GenericRecord>read()
+ *      .withBootstrapServers("broker_1:9092,broker_2:9092")
+ *      .withTopic("my_topic")
+ *      .withSchemaRegistry(""http://localhost:8081"") // Confluent Schema Registry URL
```

Comment:
nit: can you make this by default a secure connection in the example e.g. `https`

Reply:
I'd leave this example as simple as possible. All configuration options for CSR can be found in its documentation